Mathlib/RingTheory/DedekindDomain/Factorization.lean

```diff
@@ -177,3 +223,289 @@ theorem finprod_heightOneSpectrum_factorization_coe (I : Ideal R) (hI : I ≠ 0)
 #align ideal.finprod_height_one_spectrum_factorization_coe Ideal.finprod_heightOneSpectrum_factorization_coe
 
 end Ideal
+
+/-! ### Factorization of fractional ideals of Dedekind domains -/
+
+namespace FractionalIdeal
+
+open Int IsLocalization
+
+/-- If `I` is a nonzero fractional ideal, `a ∈ R`, and `J` is an ideal of `R` such that
+`I = a⁻¹J`, then `I` is equal to the product `∏_v v^(val_v(J) - val_v(a))`. -/
+theorem finprod_heightOneSpectrum_factorization {I : FractionalIdeal (nonZeroDivisors R) K}
```

Comment:
Should there be a version of this lemma in terms of `count`?

Reply:
I wrote `finprod_heightOneSpectrum_factorization'` for the version in terms of count.

Reply:
I think the `count` version should be the unprimed one?